### ISSUE TITLE
[Truffle] Fails to build due to too long filename on eCryptfs

### ISSUE THREAD

**Issue by zakkak** [oracle/graal#270]
## Environment

* OS: Ubuntu 17.10
* JDK: GraalVM 0.30.1 (JDK 8)
* graal JDK-8192814 tag
* Encrypted home directory with eCryptfs

## Error
```
/.../graal/truffle/src/com.oracle.truffle.sl/src/com/oracle/truffle/sl/runtime/SLFunctionRegistry.java:214: error: error while writing FunctionNamesHasSizeNode: /.../graal/truffle/mxbuild/src/com.oracle.truffle.sl/bin/com/oracle/truffle/sl/runtime/SLFunctionRegistry$FunctionsObject$FunctionsObjectMessageResolution$FunctionNamesObject$FunctionNamesMessageResolution$FunctionNamesHasSizeNode.class (File name too long)
```

**Comment by Gregsen** [comment]
Same on
* Ubuntu 16.04.03
* GraalVM 0.30.2

```Archiving TRUFFLE_DEBUG... [dependency com.oracle.truffle.tools updated]
/home/user/workspace/graal/truffle/src/com.oracle.truffle.sl/src/com/oracle/truffle/sl/runtime/SLFunctionRegistry.java:214: error: error while writing FunctionNamesHasSizeNode: /home/user/workspace/graal/truffle/mxbuild/src/com.oracle.truffle.sl/bin/com/oracle/truffle/sl/runtime/SLFunctionRegistry$FunctionsObject$FunctionsObjectMessageResolution$FunctionNamesObject$FunctionNamesMessageResolution$FunctionNamesHasSizeNode.class (File name too long)
                    abstract static class FunctionNamesHasSizeNode extends Node {
                                    ^
Compiling org.graalvm.compiler.phases with javac-daemon... [dependency GRAAL_OPTIONS_PROCESSOR updated]
1 error

Compiling com.oracle.truffle.sl with javac-daemon failed
1 build tasks failed
```

**Comment by nirvdrum** [comment]
We have tests in place to ensure TruffleRuby filenames work with eCryptfs, but it looks like we need something similar for Truffle itself.

As annoying as it is, one solution is to just build in a location outside of /home and symlink back to your preferred location.

**Comment by dougxc** [comment]
Can you please point to the TruffleRuby tests @nirvdrum ?
@chumer can/should Truffle be modified to respect ecryptfs file name limits?

**Comment by eregon** [comment]
The test is here: https://github.com/graalvm/truffleruby/blob/c22961aded89278cc51ffc6e93445fda3077e7b3/tool/jt.rb#L1827-L1843

**Comment by chumer** [comment]
The file length is really unnecessarily long in this case. Should be an easy fix. We should also add a check like ruby such that we don't regress.

**Comment by boris-spas** [comment]
Fixed with [7c3e1262e5022a87ad26cf504063f599da7013c6](https://github.com/oracle/graal/commit/7c3e1262e5022a87ad26cf504063f599da7013c6)
